src/platform/logging/impl/stdio/Logging.cpp

```diff
@@ -4,17 +4,64 @@
 
 #include <lib/support/logging/Constants.h>
 
+#include <inttypes.h>
 #include <stdio.h>
+#include <time.h>
+
+#if defined(__APPLE__)
+#include <pthread.h>
+#include <unistd.h>
+#elif defined(__gnu_linux__)
+#include <sys/syscall.h>
+#include <unistd.h>
+#endif
 
 namespace chip {
 namespace Logging {
 namespace Platform {
 
+// TODO: investigate whether pw_chrono or pw_log could be used here
 void LogV(const char * module, uint8_t category, const char * msg, va_list v)
 {
+#if defined(_POSIX_THREAD_SAFE_FUNCTIONS)
+    flockfile(stdout);
```

Comment:
Worth documenting that this is done because we print our log in pieces and we don't want other prints ending up in the middle of the log string, right?

Reply:
Right, I've added a comment about that.